<?xml version="1.0"?>

<launch>
  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
    
    <!-- Frame names -->
    <param name="map_frame" value="fake_map" />
    <param name="base_frame" value="petit_base_link" />
    <param name="odom_frame" value="petit_base_link" />
    
    <!-- Tf use -->
    <param name="use_tf_scan_transformation" value="true"/>
    <param name="use_tf_pose_start_estimate" value="false"/>
    <param name="pub_map_odom_transform" value="true"/>
    
    <!-- Map size / start point -->
    <param name="map_resolution" value="0.01"/>
    <param name="map_size" value="800"/>
    <param name="map_start_x" value="0.5"/>
    <param name="map_start_y" value="0.5" />
    <param name="map_multi_res_levels" value="5" />
    
    <!-- Map update parameters -->
    <param name="update_factor_free" value="0.4"/>
    <param name="update_factor_occupied" value="0.8" />    
    <param name="map_update_distance_thresh" value="0.05"/>
    <param name="map_update_angle_thresh" value="0.01" />
    
    <!-- Advertising config --> 
    <param name="advertise_map_service" value="true"/>
    
    <!-- Laser parameters -->
    <param name="laser_min_dist" value="0.13"/>
    <param name="laser_max_dist" value="4.5"/>

    <param name="map_pub_period" value="1.0"/>

    <!-- Debug parameters -->
    <!--
      <param name="output_timing" value="false"/>
      <param name="pub_drawings" value="true"/>
      <param name="pub_debug_output" value="true"/>
    -->
  </node>

  <node pkg="tf" type="static_transform_publisher" name="link_nav_broadcaster" args="0 0 0 0 0 0 odom base_link 100"/> 
  <node pkg="tf" type="static_transform_publisher" name="link_laser_neato_broadcaster" args="0.0 0 0.1 0 0 0 petit_base_link neato_laser 100"/> 
 
</launch>
  
  
